src/main/java/com/entropy/entropay/auth/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,27 @@
+package com.entropy.entropay.auth.service.impl;
+
+import com.entropy.entropay.auth.service.AuthenticationService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.authentication.DisabledException;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.stereotype.Service;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+
+    @Autowired
+    private AuthenticationManager authenticationManager;
+
+    @Override
+    public void authenticate(String username, String password) throws Exception {
+        try {
+            authenticationManager.authenticate(new UsernamePasswordAuthenticationToken(username, password));
+        } catch (DisabledException e) {
+            throw new Exception("USER_DISABLED", e);
```

Comment:
can we define at least one top-level custom exception and also have a default handler for it?